server/remote_cache/cachetools/cachetools.go

```diff
@@ -118,6 +119,90 @@ func GetBlob(ctx context.Context, bsClient bspb.ByteStreamClient, r *digest.Reso
 	return err
 }
 
+// BlobResponse is a response to an individual blob in a BatchReadBlobs request.
+type BlobResponse struct {
+	Digest *repb.Digest
+	Data   []byte
+
+	Err error
+}
+
+// BatchReadBlobs issues a BatchReadBlobs request and returns a mapping from
+// digest hash to byte payload.
+//
+// It validates the response so that if the returned err is nil, then all
+// digests in the request are guaranteed to have a corresponding map entry.
+func BatchReadBlobs(ctx context.Context, casClient repb.ContentAddressableStorageClient, req *repb.BatchReadBlobsRequest) (map[string]*BlobResponse, error) {
```

Comment:
can we return a [] of *BlobResponses instead of a map?

Reply:
Done